SourceWriters/CommonWriter.cs

```diff
@@ -0,0 +1,930 @@
+            PropertyPart += "        {\r\n";
+            int x = 0;
+            foreach (UIElement u in UserControls)
+            {
+                x++;
+                if (x != 1)
+                {
+                    PropertyPart += ",\r\n";
+                }
+                PropertyPart += "            " + u.Identifier.FirstCharToUpper();
+            }
+            PropertyPart += "\r\n";
+
+            PropertyPart += "        }\r\n";
+            PropertyPart += "\r\n";
+
+            foreach (UIElement u in UserControls)
+            {
+                if ((u.ElementType == ElementType.DropDown || u.ElementType == ElementType.RadioButtons) && !Intelli.IsEnum(u.TEnum))
+                {
+                    string identifier = u.Identifier.FirstCharToUpper();
+                    PropertyPart += "        public enum " + identifier + "Options\r\n";
+                    PropertyPart += "        {\r\n";
+                    int z = 0;
+                    foreach (string Option in u.ToOptionArray())
+                    {
+                        z++;
+                        if (z != 1)
+                        {
+                            PropertyPart += ",\r\n";
+                        }
+                        PropertyPart += "            " + identifier + "Option" + z.ToString();
+                    }
+                    PropertyPart += "\r\n";
+                    PropertyPart += "        }\r\n";
+                    PropertyPart += "\r\n";
+                }
+            }
+
+            // if we have a random number generator control, include the following lines...
+            if (UserControls.Any(u => u.ElementType == ElementType.ReseedButton))
+            {
+                PropertyPart += "        [ThreadStatic]\r\n";
+                PropertyPart += "        private static Random RandomNumber;\r\n";
+                PropertyPart += "\r\n";
+                PropertyPart += "        private int randomSeed;\r\n";
+                PropertyPart += "        private int instanceSeed;\r\n";
+                PropertyPart += "\r\n";
+            }
+
+            // generate OnCreatePropertyCollection()
+            PropertyPart += "\r\n";
+
+            if (isFileType)
+            {
+                PropertyPart += "        public override PropertyCollection OnCreateSavePropertyCollection()\r\n";
+            }
+            else
+            {
+                PropertyPart += "        protected override PropertyCollection OnCreatePropertyCollection()\r\n";
+            }
+
+            PropertyPart += "        {\r\n";
+
+            // Check to see if we're including a color wheel without an alpha slider
+            if (UserControls.Any(u => u.ElementType == ElementType.ColorWheel && !u.ColorWheelOptions.HasFlag(ColorWheelOptions.Alpha)))
+            {
+                PropertyPart += "            ColorBgra PrimaryColor = EnvironmentParameters.PrimaryColor.NewAlpha(byte.MaxValue);\r\n";
```

Comment:
for a `BitmapEffect`, the `Environment.PrimaryColor` will be of type `ColorBgra32`, which does not have a `NewAlpha` method.

You can do this instead though:

`Environment.PrimaryColor with { A = byte.MaxValue }`